app/src/main/java/org/oppia/app/FakeDataProvider.kt

```diff
@@ -0,0 +1,27 @@
+package org.oppia.app
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import org.oppia.data.backends.gae.model.GaeExplorationContainer
+
+// Remove this file at later stage once domain and model module are working.
```

Comment:
I actually think we should be faking data at a higher level than this (e.g. in domain logic controllers). We should spec out those interfaces and submit them with stubs that return structured fake data, then directly rely on those for the UI.

What fake data do we need for this PR? Can we just check in the structure and hook up to domain controllers in a subsequent PR?

Reply:
Actually, I created this file but we are not using it anywhere, so we can remove this right now.

But still there is one issue, we will be using models which are present in `data` module like `GaeSubtitledHtml` `GaeState` etc. in subsequent PRs and as far as I understand the Technical Document, we should not be interacting directly with `data` module from `app` module. So for that case I think `domain` module will need something similar to models defined in `data` module.

Any comments or suggestions on this?

Reply:
If this is for testing purposes, we can use the mock GAE services directly in tests to influence the data returned by domain controllers. We can extend those services to actually set up state rather than returning the same fixed data.

If this isn't for tests, app code does not need to rely on GaeSubtitledHtml, etc. since it will receive data via the domain layer defined in the model module (see how topic list data is piped in #106).

Does this help clarify?